### PR TITLE
net: lwm2m: Deprecate Kconfig for LwM2M RD Client

### PR DESCRIPTION
**Pull request by juhaylinen**
RD-Client is essential part of LwM2M specification and it can't be disabled from LwM2M engine. This commit removes Kconfig variable CONFIG_LWM2M_RD_CLIENT_SUPPORT and all usages if it.

Signed-off-by: Juha Ylinen <juha.ylinen@nordicsemi.no>

### CODE REVIEW

File: subsys/net/lib/lwm2m/Kconfig [hunk 1]
```diff
@@ -192,13 +192,12 @@ config LWM2M_TLS_SESSION_CACHING
...
 	help
 	  This config suspend socket handler which skip socket polling process.
 
 config LWM2M_RD_CLIENT_SUPPORT
 	bool "support for LWM2M client bootstrap/registration state machine"
-	default y
+	select DEPRECATED
 	help
 	  Client will use registration state machine to locate and connect to
 	  LWM2M servers (including bootstrap server support)
```
Comment (SeppoTakalo) [inline subsys/net/lib/lwm2m/Kconfig:203]:
```
config LWM2M_RD_CLIENT_SUPPORT
	bool "support for LWM2M client bootstrap/registration state machine"
	default n
	select DEPRECATED
        help
             Deprecated flag. RD state machine is always part of engine. It cannot be disabled.
```

**Review by SeppoTakalo — DISMISSED** [review]
Looks OK.
I wonder if the Kconfig help could also tell that this is deprecated.

**Review by SeppoTakalo — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]
LGTM

**Comment by carlescufi** [comment]
@juhaylinen please rebase

**Review by rlubos — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Pull request merged**